routing/record/validation.go

```diff
@@ -73,13 +73,19 @@ func (v Validator) IsSigned(k key.Key) (bool, error) {
 // verifies that the passed in record value is the PublicKey
 // that matches the passed in key.
 func ValidatePublicKeyRecord(k key.Key, val []byte) error {
-	keyparts := bytes.Split([]byte(k), []byte("/"))
-	if len(keyparts) < 3 {
-		return errors.New("invalid key")
+	if len(k) != 38 {
+		return errors.New("invalid public key record key")
```

Comment:
If we don't want to do that we can do:
```go
bytes.SplitN([]byte(k). []byte("/"), 2)
```
then the second part should contain the rest of the key (even if it contains more slashes).